python/tests/test_lowlevel.py

```diff
@@ -489,6 +497,30 @@ def test_direct_instantiation(self):
         assert "IBD" in output
 
 
+class TestIbdSegmentList:
+    def test_direct_instantiation(self):
+        # Nobody should do this, but just in case
+        seglist = _tskit.IbdSegmentList()
+        attrs = ["num_segments", "total_span", "left", "right", "node"]
+        for attr in attrs:
+            with pytest.raises(SystemError, match="not initialised"):
+                getattr(seglist, attr)
+
+    def test_memory_management(self):
```

Comment:
I don't understand what this test is doing -- can you explain a bit?

Reply:
It's basically the same as the previous comment - we're checking some tricky details about how memory is managed, making sure we don't leak or segfault in odd situations. In this case, the ownership of the underlying memory is being managed by the object that we ``del``, and we're making sure that this object doesn't actually disappear until after all the things that reference it also disappear.

Some background on [reference counting](https://towardsdatascience.com/understanding-reference-counting-in-python-3894b71b5611) in Python might clear things up a bit, if you're interested.